src/main/java/tech/aroma/data/sql/SQLMessageRepository.java

```diff
@@ -0,0 +1,258 @@
+
+    @Inject
+    SQLMessageRepository(JdbcTemplate database, DatabaseSerializer<Message> serializer)
+    {
+        checkThat(database, serializer)
+                .are(notNull());
+
+        this.database = database;
+        this.serializer = serializer;
+    }
+
+    @Override
+    public void saveMessage(Message message, @Optional LengthOfTime lifetime) throws TException
+    {
+        checkThat(message)
+                .throwing(InvalidArgumentException.class)
+                .is(notNull())
+                .is(validMessage());
+
+        Duration messageDuration = null;
+
+        if (Objects.nonNull(lifetime))
+        {
+
+            checkThat(lifetime)
+                    .throwing(InvalidArgumentException.class)
+                    .is(validLengthOfTime());
+
+            messageDuration = TimeFunctions.lengthOfTimeToDuration().apply(lifetime);
+        }
+
+        _saveMessage(message, messageDuration);
+    }
+
+    private void _saveMessage(Message message, Duration messageDuration) throws OperationFailedException
+    {
+        final String statement = SQLStatements.Inserts.MESSAGE;
+
+        try
+        {
+            serializer.save(message, messageDuration, statement, database);
+        }
+        catch (Exception ex)
+        {
+            LOG.error("Failed to serialize Message {} using statement [{}]", message, statement, ex);
+            throw new OperationFailedException(ex.getMessage());
+        }
+    }
+
+    @Override
+    public Message getMessage(String applicationId, String messageId) throws TException
+    {
+        checkThat(applicationId, messageId)
+                .throwing(InvalidArgumentException.class)
+                .are(validUUID());
+
+        final UUID appId = UUID.fromString(applicationId);
+        final UUID msgId = UUID.fromString(messageId);
+        final String statement = SQLStatements.Queries.SELECT_MESSAGE;
+
+        Message message;
+
+        try
+        {
+            message = database.queryForObject(statement, serializer, appId, msgId);
+
+        }
+        catch (EmptyResultDataAccessException ex)
+        {
+            LOG.warn("Message does not exist: [{}/{}]", appId, messageId, ex);
+            throw new DoesNotExistException(ex.getMessage());
+        }
+        catch (Exception ex)
+        {
+            LOG.error("Failed to get message [{}/{}]", applicationId, messageId, ex);
+            throw new OperationFailedException(ex.getMessage());
+        }
+
+        checkThat(message)
+                .throwing(DoesNotExistException.class)
+                .is(notNull());
+
+        return message;
+    }
+
+    @Override
+    public void deleteMessage(String applicationId, String messageId) throws TException
+    {
+        checkThat(applicationId, messageId)
+                .throwing(InvalidArgumentException.class)
+                .are(validUUID());
+
+        UUID appId = UUID.fromString(applicationId);
```

Comment:
Try testing this without string conversions to see if the tests pass.
I think the database does this conversion for us.